emulated_hue/api.py

```diff
@@ -643,14 +657,63 @@ async def __async_light_action(self, entity: dict, request_data: dict) -> None:
         if const.HUE_ATTR_TRANSITION in request_data:
             # Duration of the transition from the light to the new state
             # is given as a multiple of 100ms and defaults to 4 (400ms).
-            transitiontime = request_data[const.HUE_ATTR_TRANSITION] / 10
+            if request_data[const.HUE_ATTR_TRANSITION] * 100 <= throttle_ms:
+                transitiontime = throttle_ms / 1000
+            else:
+                transitiontime = request_data[const.HUE_ATTR_TRANSITION] / 10
             data[const.HASS_ATTR_TRANSITION] = transitiontime
         else:
-            data[const.HASS_ATTR_TRANSITION] = 0.4
+            data[const.HASS_ATTR_TRANSITION] = (
+                0.4 if throttle_ms <= 400 else throttle_ms / 1000
+            )
 
         # execute service
         await self.hass.async_call_service(const.HASS_DOMAIN_LIGHT, service, data)
 
+    def __update_allowed(
+        self, entity: dict, light_data: dict, throttle_ms: int
+    ) -> bool:
+        """Minimalistic form of throttling, only allow updates to a light within a timespan."""
+
+        prev_data = self._prev_data.get(entity["entity_id"], {})
```

Comment:
We need to just always return true either here or before this is called if the throttle is set to 0, skipping this check.

Reply:
update_allowed function can return True early if:

- current light state is same as command values, e.g. no need to send brightness 50 if that is already the case.
- throttle setting is set to 0
- throttle is set to a value in ms and previous update was more than that ms ago

Reply:
I'v moved if not throttle_ms: return True here as requested. All 3 scenarios now is implemented (comparing light states is very basic, but I think that this should be reworked when controllers will be implemented - it will equire stooring ich state attribute separately to compare previous one to current one - no all messages will have all attributes...)